### PR TITLE
release 23.4 notes

### PR DESCRIPTION
**Pull request by adwk67**
Release notes for SDP 23.4

N.B. check the text in the `stackablectl release list` output (should match https://github.com/stackabletech/release/pull/22/files)
N.B. check correctness of deleting CRDs rather than applying them
N.B. check links that assume "stable"

### CODE REVIEW

File: modules/ROOT/pages/release_notes.adoc [hunk 1]
```diff
@@ -3,6 +3,367 @@
...
+
+Log Aggregation::
+
+The logging framework (added to the platform in Release 23.1) that offers a consistent custom resource configuration and a separate, persisted sink, has been rolled out across all operators. See this https://docs.stackable.tech/home/stable/contributor/adr/ADR023-product-image-selection.html[ADR] and this https://docs.stackable.tech/home/stable/concepts/product_image_selection.html[concepts page] for more information.
+
+Service Type::
+
+The Service type can now be specified: this enables a later switch to using ListenerClasses for the exposure of Services. This change is breaking: for security reasons the default is set to the cluster-internal ListenerClass. A cluster can be made accessible from outside of Kubernetes by setting clusterConfig.listenerClass to external-unstable or external-stable.
```
Comment (maltesander) [inline modules/ROOT/pages/release_notes.adoc:27]:
```suggestion
The Service type can now be specified in all products. This currently differences between the internal ClusterIP and the external NodePort and is forward compatible with the [ListenerClass](https://docs.stackable.tech/home/stable/listener-operator/listenerclass.html) for the automatic exposure of Services via the Listener Operator. This change is breaking. For security reasons the default is set to the cluster-internal (ClusterIP) ListenerClass. A cluster can be made accessible from outside of Kubernetes by setting clusterConfig.listenerClass to external-unstable (NodePort) or external-stable (LoadBalancer).
```

File: modules/ROOT/pages/release_notes.adoc [hunk 2]
```diff
@@ -3,6 +3,367 @@
...
+== Release 23.4
+
+The focus in this platform release is on the support of default/custom affinities and the status field, as well as the rollout of log aggregation across the remaining operators. Additionally, all operators have been adapted and tested for compatibility with Openshift clusters (versions 4.10 and 4.11). Several operators from the 23.1 platform release were already certified against Openshift: over the coming weeks this certification process will now be extended across all operators and will be specific to each platform-release.
+
+=== New/extended platform features
+The following new major platform features were added:
+
+Status Field::
+
+Operators of the Stackable Data Platform create, manage and delete Kubernetes resources: in order to easily query the health state of the products - and react accordingly - Stackable Operators use the several predefined condition types to capture different aspects of a product's availability. See this https://docs.stackable.tech/home/stable/contributor/adr/ADR027-status[ADR] for more information.
```
[the remark this answers is not in the document]
Comment (adwk67) [reply modules/ROOT/pages/release_notes.adoc:15]: That should work when the docs are taken live.

File: modules/ROOT/pages/release_notes.adoc [hunk 3]
```diff
@@ -3,6 +3,367 @@
...
+=== Supported Kubernetes versions
+This release supports the following Kubernetes versions:
+
+* `1.26`
+* `1.25`
+* `1.24`
+* `1.23`
+
+==== Breaking changes
+You will need to adapt your existing CRDs due to the following breaking changes detailed below.
```
[the remark this answers is not in the document]
Comment (adwk67) [reply modules/ROOT/pages/release_notes.adoc:73]: Yes, good point. I'll leave 1.23 in but qualify its support.

File: modules/ROOT/pages/release_notes.adoc [hunk 4]
```diff
@@ -3,6 +3,373 @@
 The Stackable platform consists of multiple operators that work together.
 Periodically a platform release is made, including all components of the platform at a specific version.
 
+== Release 23.4
+
+The focus in this platform release is on the support of default/custom affinities and the status field, as well as the rollout of log aggregation across the remaining operators. Additionally, all operators have been adapted and tested for compatibility with Openshift clusters (versions 4.10 and 4.11). Several operators from the 23.1 platform release were already certified against Openshift. Over the coming weeks this certification process will now be extended across all operators and will be specific to each platform-release.
```
Comment (razvan) [inline modules/ROOT/pages/release_notes.adoc:8]: I removed the "what we will do in the future" stuff from here.

```suggestion
The focus in this platform release is on the support of default/custom affinities and the status field, as well as the rollout of log aggregation across the remaining operators. Additionally, all operators have been updated and tested for compatibility with OpenShift clusters (versions 4.10 and 4.11). Several operators from the 23.1 platform release were already certified against OpenShift.
```

File: modules/ROOT/pages/release_notes.adoc [hunk 5]
```diff
@@ -3,6 +3,373 @@
...
+=== New / extended platform features
+The following new major platform features were added:
+
+Cluster Operation::
+
+The first part of https://docs.stackable.tech/home/stable/concepts/cluster_operations.html[ClusterOperations] was rolled out in every applicable Stackable Operator. This supports pausing the cluster reconciliation and stopping the cluster completely. Pausing reconciliation will not apply any changes to the Kubernetes resources (e.g. when changing the custom resource). Stopping the cluster will set all replicas of StatefulSets, Deployments or DaemonSets to zero and therefore deleting all Pods belonging to that cluster (not the PVCs).
+
+Status Field::
+
+Operators of the Stackable Data Platform create, manage and delete Kubernetes resources: in order to easily query the health state of the products - and react accordingly - Stackable Operators use the several predefined condition types to capture different aspects of a product's availability. See this https://docs.stackable.tech/home/stable/contributor/adr/ADR027-status[ADR] for more information.
```
Comment (razvan) [inline modules/ROOT/pages/release_notes.adoc:19]:
```suggestion
Operators of the Stackable Data Platform create, manage and delete Kubernetes resources: in order to easily query the health state of the products - and react accordingly - Stackable Operators use several predefined condition types to capture different aspects of a product's availability. See this https://docs.stackable.tech/home/stable/contributor/adr/ADR027-status[ADR] for more information.
```

File: modules/ROOT/pages/release_notes.adoc [hunk 6]
```diff
@@ -3,6 +3,373 @@
...
+
+Default / Custom Affinities::
+
+In Kubernetes there are different ways to influence how Pods are assigned to Nodes. In some cases it makes sense to co-locate certain services that communicate a lot with each other, such as HBase regionservers with HDFS datanodes. In other cases it makes sense to distribute the Pods among as many Nodes as possible. There may also be additional requirements e.g. placing important services - such as HDFS namenodes - in different racks, datacenter rooms or even datacenters. This release supports default affinities that should suffice for many scenarios out-of-the box, as well as enabling the configuration of custom affinity rules at a role and/or role-group level. See this https://docs.stackable.tech/home/stable/contributor/adr/ADR026-affinities[ADR] for more information.
```
Comment (razvan) [inline modules/ROOT/pages/release_notes.adoc:23]:
```suggestion
In Kubernetes there are different ways to influence how Pods are assigned to Nodes. In some cases it makes sense to co-locate certain services that communicate a lot with each other, such as HBase regionservers with HDFS datanodes. In other cases it makes sense to distribute the Pods among as many Nodes as possible. There may also be additional requirements e.g. placing important services - such as HDFS namenodes - in different racks, datacenter rooms or even datacenters. This release implements default affinities that should suffice for many scenarios out-of-the box, while also allowing for custom affinity rules at a role and/or role-group level. See this https://docs.stackable.tech/home/stable/contributor/adr/ADR026-affinities[ADR] for more information.
```

File: modules/ROOT/pages/release_notes.adoc [hunk 7]
```diff
@@ -3,6 +3,373 @@
...
+
+Log Aggregation::
+
+The logging framework (added to the platform in Release 23.1) offers a consistent custom resource configuration and a separate, persisted sink (the default implementation is Opensearch but is interchangeable). This has now been rolled out across all operators. See this https://docs.stackable.tech/home/stable/contributor/adr/adr025-logging_architecture[ADR] and this https://docs.stackable.tech/home/stable/concepts/logging.html[concepts page] for more information.
```
Comment (razvan) [inline modules/ROOT/pages/release_notes.adoc:27]:
```suggestion
The logging framework (added to the platform in Release 23.1) offers a consistent custom resource configuration and a separate, persisted sink (defaulting to OpenSearch). This has now been rolled out across all products. See this https://docs.stackable.tech/home/stable/contributor/adr/adr025-logging_architecture[ADR] and this https://docs.stackable.tech/home/stable/concepts/logging.html[concepts page] for more information.
```

File: modules/ROOT/pages/release_notes.adoc [hunk 8]
```diff
@@ -3,6 +3,373 @@
...
+
+Service Type::
+
+The Service type can now be specified in all products. This currently differentiates between the internal ClusterIP and the external NodePort and is forward compatible with the [ListenerClass](https://docs.stackable.tech/home/stable/listener-operator/listenerclass.html) for the automatic exposure of Services via the Listener Operator. This change is breaking. For security reasons the default is set to the cluster-internal (ClusterIP) ListenerClass. A cluster can be made accessible from outside of Kubernetes by setting clusterConfig.listenerClass to external-unstable (NodePort) or external-stable (LoadBalancer).
```
Comment (razvan) [inline modules/ROOT/pages/release_notes.adoc:31]:
```suggestion
The Service type can now be specified in all products. This currently differentiates between the internal ClusterIP and the external NodePort and is forward compatible with the [ListenerClass](https://docs.stackable.tech/home/stable/listener-operator/listenerclass.html) for the automatic exposure of Services via the Listener Operator. This change not backwards compatible with older platform releases. For security reasons, the default is set to the cluster-internal (ClusterIP) ListenerClass. A cluster can be exposed outside of Kubernetes by setting clusterConfig.listenerClass to external-unstable (NodePort) or external-stable (LoadBalancer).
```

File: modules/ROOT/pages/release_notes.adoc [hunk 9]
```diff
@@ -3,6 +3,373 @@
...
+
+
+=== Supported Kubernetes versions
+
+This release supports the following Kubernetes versions:
+
+* `1.26`
+* `1.25`
+* `1.24`
+* `1.23` (it is planned to discontinue support for this version in the next release)
```
Comment (razvan) [inline modules/ROOT/pages/release_notes.adoc:75]: Maybe also list OpenShift 4.10 and 4.11 now?

**Review by razvan — COMMENTED** [review]
lgtm. some suggestions

**Pull request merged**